### ISSUE TITLE
Building on windows and mingw

### ISSUE THREAD

**Issue by divinon** [godlikepanos/anki-3d-engine#46]
Hello, I've some feedback on building on Windows with mingw-w64.

The CMAKE_BUILD_TYPE check could also check if the var is empty like this:
`if(NOT DEFINED CMAKE_BUILD_TYPE OR "${CMAKE_BUILD_TYPE}" STREQUAL "")`
otherwise in that case, CMAKE_BUILD_TYPE defined but empty, cmake fails.

Could not compile 'version.rc' for SDL2 because of this:
`-DIMGUI_USER_CONFIG="<anki/ui/ImGuiConfig.h>"`
Removing `add_definitions(-DIMGUI_USER_CONFIG=<anki/ui/ImGuiConfig.h>)` from cmake fixed it.

Could not find 'vulkan-1' for linking. The files 'linklibs.rsp' have '-LX:/path/to/VulkanSDK/Bin' but my libraries from Lunarg's Vulkan SDK are in '-LX:/path/to/VulkanSDK/Lib', not sure how to fix it in cmake, so I've tampered the rsp files.
My env vars VK_SDK_PATH and VULKAN_SDK point to 'X:/path/to/VulkanSDK'.

Finally, an error on execution:
`[E][RSRC] shaders/ClearTextureCompute.glslp: Malformed expression: #pragma anki (...ShaderProgramPreProcessor.cpp:755 parsePragmaStart)`
Which is quite strange because the full line is '`#pragma anki start comp`'. The line seems truncated.

Unfortunately I cannot debug because my biggest problem now is the linking time with mingw-make (in debug, I haven't tried release), it is over 6 minutes, the exe files are in fact rather huge (650Mb). Do you have any suggestion?

**Comment by divinon** [comment]
Tried enabling ANKI_LTO and I got a lot of undefined reference to glslang functions.
Tried enabling ANKI_STRIP and moved the project to a FAT32 file system. Now the linker doesn't hang anymore, before on a NTFS I could see ld.exe stuck for about 2 minutes (no movement in the memory or on the files), this problem of mingw's ld.exe and big files on a NTFS goes back to 2011, for reference I'm using MinGW-W64-builds-4.3.5.
Now I'm at 75seconds, a lot better but still a bother. The library 'libankid.a' is 149 Mbyte, but 'libSPIRV-Tools-optd.a' is about 490 Mbyte, isn't it too big?

**Comment by godlikepanos** [comment]
Thanks for the detailed report! mingw was always problematic and I'm not surprised with your findings. 

Do you want me to investigate the mingw issues or provide documentation on how to build with MSVC? The 2nd option might be more forward looking but MSVC has a few C++ issues that I need to workaround.

**Comment by divinon** [comment]
Oh thanks but this was only a test, so don't waste your time if it wasn't in your plans. I was just curious about that hefty libSPIRV-Tools.
I've tried VS2017 (I prefer gcc and mingw but to be pragmatic most of the people on Windows use msvc), I was able to change the ANKI_ENABLE_METHOD macro, but then a bunch of other errors were above my zero template magic skills to follow. So that's it, I hope it was useful somehow.
